src/NumericalAlgorithms/LinearOperators/PartialDerivatives.cpp

```diff
@@ -98,42 +100,70 @@ void logical_partial_derivative(
 
   set_number_of_grid_points(logical_derivative_of_u,
                             mesh.number_of_grid_points());
-  const Matrix empty_matrix{};
-  std::array<std::reference_wrapper<const Matrix>, Dim> diff_matrices{
-      make_array<Dim, std::reference_wrapper<const Matrix>>(empty_matrix)};
-  for (size_t d = 0; d < Dim; ++d) {
-    gsl::at(diff_matrices, d) =
-        std::cref(Spectral::differentiation_matrix(mesh.slice_through(d)));
-  }
+  if (Dim == 3 and mesh.basis(1) == Spectral::Basis::SphericalHarmonic) {
```

Comment:
Note that logical coordinates in our 3D wedges are ordered [theta, phi, r]. I think it would be helpful to align the radial dimension of neighboring cubed-wedges and spherical shells, i.e., assume in this function that the radial dimension is `zeta` (and put this in the documentation of the function). Otherwise we have to deal with this misalignment in a bunch of places in the code, e.g. when we choose p-refinement in domain creators.

Reply:
I extremely strongly disagree, anything other than `r, theta, phi` is unintuitive and will cause problems. If anything gets changed, it should be the wedges that get changed, but that is beyond the scope of this PR.

Reply:
Sure we can change the wedges (I also think that would be the better solution). Having cubed shells and spherical shells have a different logical radial axis will be a nightmare though, so I'm just pointing out here that we have to fix that...